flink-table/flink-table-common/src/main/java/org/apache/flink/table/typeutils/TimeIndicatorTypeInfo.java

```diff
@@ -62,13 +61,10 @@ protected TimeIndicatorTypeInfo(boolean isEventTime) {
         this.isEventTime = isEventTime;
     }
 
-    // this replaces the effective serializer by a LongSerializer
-    // it is a hacky but efficient solution to keep the object creation overhead low but still
-    // be compatible with the corresponding SqlTimestampTypeInfo
     @Override
     @SuppressWarnings("unchecked")
     public TypeSerializer<Timestamp> createSerializer(ExecutionConfig executionConfig) {
-        return (TypeSerializer) LongSerializer.INSTANCE;
+        return (TypeSerializer) LocalDateTimeSerializer.INSTANCE;
```

Comment:
Can you explain why we need to modify this? Does it have any other impact?

Reply:
The problem has stucked me for two weeks, if not do so, some tests like `FunctionITCase#testAggregateFunction` would fail. In fact, `TimeIndicatorTypeInfo` as a deprecated class just provide a hacky solution and not cover the new framework, when deprecated `TableSchema` coverted from new `ResolvedSchema`, extra info like `rowtime` would retain, thus flink regard a timestamp watermark column as `TimeIndicatorTypeInfo` and use LongSerializer to serialize a LocalDateTime object then throw ClassCastException.

Reply:
can we improve the conversion logic of `ResolvedSchema to TableSchema` ?

Reply:
I used to remove `rowtime` attri during the  conversion from `ResolvedSchema` to `Schema`, but this will make about fifty test cases trigger the Calcite exception as follow:
```
Cannot add expression of different type to set:
set type is RecordType(INTEGER amount, VARCHAR(2147483647) CHARACTER SET "UTF-16LE" currency, TIMESTAMP(3) *ROWTIME* rowtime, TIMESTAMP_LTZ(3) *PROCTIME* NOT NULL proctime, VARCHAR(2147483647) CHARACTER SET "UTF-16LE" NOT NULL currency0, INTEGER rate, TIMESTAMP(3) rowtime0) NOT NULL
expression type is RecordType(INTEGER amount, VARCHAR(2147483647) CHARACTER SET "UTF-16LE" currency, TIMESTAMP(3) *ROWTIME* rowtime, TIMESTAMP_LTZ(3) *PROCTIME* NOT NULL proctime, VARCHAR(2147483647) CHARACTER SET "UTF-16LE" NOT NULL currency0, INTEGER rate, TIMESTAMP(3) *ROWTIME* rowtime0) NOT NULL
```